src/pages/api/webhooks/webhook.ts

```diff
@@ -0,0 +1,39 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+interface IssuePayload {
+  issue: {
+    html_url: string;
+    title: string;
+  };
+  repository: {
+    name: string;
+  };
+}
+
+export default function webhooksHandler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === 'POST') {
+    const eventType = req.headers['x-github-event'];
+    if (eventType !== 'issues') {
+      console.log(`Webhook ignored. Unsupported event type: ${eventType as string}`);
+      res.status(200).send('Webhook ignored');
+      return;
+    }
+
+    const payload = req.body as IssuePayload;
```

Comment:
What happens if this cast fails?

Reply:
Yeah this is a good place to use zod for example / isn't trpc supposed to help here?

Reply:
As I receive a real webhook hit from either GitHub or Google and not a command I created on my own, I'm sure this cast will definitely fail and that is when I assess what needs to be changed. This file should've been in another PR but the purpose of this webhook file is to just have a basic setup of a webhook on the server.

Reply:
> Yeah this is a good place to use zod for example / isn't trpc supposed to help here?

Honestly, I'll be working on this now but I believe so yeah. My initial thought is to wait until I get a hit from GitHub or Google and once that occurs, I use tRPC to check if one of the automations from the current logged in user matches with what data was received. If so, then send an email, cancel the meeting, etc...